packages/asset-uploader/src/types.ts

```diff
@@ -0,0 +1,21 @@
+import { z } from "zod";
+import { NodeOnDiskFile } from "@remix-run/node";
+
+const SingleImageInUpload = z.instanceof(NodeOnDiskFile);
+
+export const ImagesUpload = z.array(SingleImageInUpload);
+
+export type ImagesUpload = z.infer<typeof ImagesUpload>;
+export type SingleImageInUpload = z.infer<typeof SingleImageInUpload>;
+
+export const ImagesUploadedSuccess = z.object({
+  Location: z.string(),
+});
+export type ImagesUploadedSuccess = z.infer<typeof ImagesUploadedSuccess>;
+
+export const s3EnvVariables = z.object({
+  S3_ENDPOINT: z.string(),
+  S3_REGION: z.string(),
+  S3_ACCESS_KEY_ID: z.string(),
+  S3_SECRET_ACCESS_KEY: z.string(),
```

Comment:
Oh I missed this one, so the thing is its not all variables, there are some optional, you can use z.optional or something for this and then you can use just the return from schema.parse and never have to deal with accessing process.env again, have guaranteed values

Reply:
all done!